teamcity: failed test: Example_demo_locality

The following tests appear to have failed on master (testrace): Example_demo_locality

You may want to check [for open issues](https://github.com/cockroachdb/cockroach/issues?q=is%3Aissue+is%3Aopen+Example_demo_locality).

[#1504636](https://teamcity.cockroachdb.com/viewLog.html?buildId=1504636):


```
Example_demo_locality
...rc/github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2/frame.go:237 +0xad
github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2.(*Framer).ReadFrame(0xc0104460e0, 0xc00dcb4720, 0xc00dcb4720, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2/frame.go:492 +0xfc
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/internal/transport.(*http2Server).HandleStreams(0xc00dfa1800, 0xc00f841da0, 0x4c93730)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/internal/transport/http2_server.go:431 +0xdb
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveStreams(0xc0040b1980, 0x597c580, 0xc00dfa1800)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:713 +0x171
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).handleRawConn.func1(0xc0040b1980, 0x597c580, 0xc00dfa1800)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:675 +0x51
created by github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).handleRawConn
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:674 +0x771

goroutine 137671 [semacquire]:
sync.runtime_SemacquireMutex(0xc00fab81f4, 0x6c00000000)
	/usr/local/go/src/runtime/sema.go:71 +0x3d
sync.(*Mutex).Lock(0xc00fab81f0)
	/usr/local/go/src/sync/mutex.go:134 +0x1e2
sync.(*RWMutex).Lock(0xc00fab81f0)
	/usr/local/go/src/sync/rwmutex.go:93 +0x4a
github.com/cockroachdb/cockroach/pkg/util/syncutil.(*RWMutex).Lock(0xc00fab81f0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/syncutil/mutex_sync_race.go:62 +0x3b
github.com/cockroachdb/cockroach/pkg/rpc.(*RemoteClockMonitor).VerifyClockOffset(0xc00fab81e0, 0x592b900, 0xc00dadb500, 0xed51c7685, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/clock_offset.go:202 +0x10f
github.com/cockroachdb/cockroach/pkg/server.NewServer.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:259 +0xbd
github.com/cockroachdb/cockroach/pkg/rpc.(*Context).runHeartbeat.func2(0x592b840, 0xc00b9c5c00, 0xc0031ef900, 0x4b20099)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:1014 +0xba1
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc0031ef900, 0x592b840, 0xc00b9c5c00, 0x4b20099, 0xd, 0xc01f301dd8, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:300 +0xf9
github.com/cockroachdb/cockroach/pkg/rpc.(*Context).runHeartbeat(0xc002587380, 0xc011f5cd80, 0xc002539500, 0x14, 0xc00f7432c0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:942 +0x57c
github.com/cockroachdb/cockroach/pkg/rpc.(*Context).grpcDialNodeInternal.func1.1.1(0x592b900, 0xc0098ee270)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:878 +0x95
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc009b910b0, 0xc0031ef900, 0xc00e940ff0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x160
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:189 +0xc4

goroutine 138124 [select]:
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).Start.func3(0x592b900, 0xc00c39a060)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:322 +0x170
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc000f033f0, 0xc0031ef900, 0xc009023fb0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x160
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:189 +0xc4

goroutine 137769 [select, 37 minutes]:

goroutine 140761 [IO wait]:
	/usr/local/go/src/runtime/netpoll.go:182 +0x56



***** Text was not loaded fully because its' size exceeds 2 MB, see full log for the whole text *****





```

Please assign, take a look and update the issue accordingly.

It looks like a timeout?

```
Example_demo_locality (1s)
[17:39:36][Test Output]
[demo --nodes 3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
[demo --nodes 9 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id]
panic: test timed out after 45m0s
```

--- a/pkg/cli/cli_test.go
+++ b/pkg/cli/cli_test.go
@@ -470,45 +470,6 @@ func Example_logging() {
 	// 1
 }
 
-func Example_demo_locality() {
-	c := newCLITest(cliTestParams{noServer: true})
-	defer c.cleanup()
-
-	testData := [][]string{
-		{`demo`, `--nodes`, `3`, `-e`, `select node_id, locality from crdb_internal.gossip_nodes order by node_id`},
-		{`demo`, `--nodes`, `9`, `-e`, `select node_id, locality from crdb_internal.gossip_nodes order by node_id`},
-		{`demo`, `--nodes`, `3`, `--demo-locality=region=us-east1:region=us-east2:region=us-east3`,
-			`-e`, `select node_id, locality from crdb_internal.gossip_nodes order by node_id`},
-	}
-	setCLIDefaultsForTests()
-	for _, cmd := range testData {
-		c.RunWithArgs(cmd)
-	}
-
-	// Output:
-	// demo --nodes 3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
-	// node_id	locality
-	// 1	region=us-east1,az=b
-	// 2	region=us-east1,az=c
-	// 3	region=us-east1,az=d
-	// demo --nodes 9 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
-	// node_id	locality
-	// 1	region=us-east1,az=b
-	// 2	region=us-east1,az=c
-	// 3	region=us-east1,az=d
-	// 4	region=us-west1,az=a
-	// 5	region=us-west1,az=b
-	// 6	region=us-west1,az=c
-	// 7	region=europe-west1,az=b
-	// 8	region=europe-west1,az=c
-	// 9	region=europe-west1,az=d
-	// demo --nodes 3 --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
-	// node_id	locality
-	// 1	region=us-east1
-	// 2	region=us-east2
-	// 3	region=us-east3
-}
-
 func Example_demo() {
 	c := newCLITest(cliTestParams{noServer: true})
 	defer c.cleanup()

--- a/pkg/cli/demo_locality_test.go
+++ b/pkg/cli/demo_locality_test.go
@@ -1,0 +1,52 @@
+// Copyright 2019 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+// +build !race
+
+package cli
+
+func Example_demo_locality() {
+	c := newCLITest(cliTestParams{noServer: true})
+	defer c.cleanup()
+
+	testData := [][]string{
+		{`demo`, `--nodes`, `3`, `-e`, `select node_id, locality from crdb_internal.gossip_nodes order by node_id`},
+		{`demo`, `--nodes`, `9`, `-e`, `select node_id, locality from crdb_internal.gossip_nodes order by node_id`},
+		{`demo`, `--nodes`, `3`, `--demo-locality=region=us-east1:region=us-east2:region=us-east3`,
+			`-e`, `select node_id, locality from crdb_internal.gossip_nodes order by node_id`},
+	}
+	setCLIDefaultsForTests()
+	for _, cmd := range testData {
+		c.RunWithArgs(cmd)
+	}
+
+	// Output:
+	// demo --nodes 3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
+	// node_id	locality
+	// 1	region=us-east1,az=b
+	// 2	region=us-east1,az=c
+	// 3	region=us-east1,az=d
+	// demo --nodes 9 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
+	// node_id	locality
+	// 1	region=us-east1,az=b
+	// 2	region=us-east1,az=c
+	// 3	region=us-east1,az=d
+	// 4	region=us-west1,az=a
+	// 5	region=us-west1,az=b
+	// 6	region=us-west1,az=c
+	// 7	region=europe-west1,az=b
+	// 8	region=europe-west1,az=c
+	// 9	region=europe-west1,az=d
+	// demo --nodes 3 --demo-locality=region=us-east1:region=us-east2:region=us-east3 -e select node_id, locality from crdb_internal.gossip_nodes order by node_id
+	// node_id	locality
+	// 1	region=us-east1
+	// 2	region=us-east2
+	// 3	region=us-east3
+}

